misc/buildkite/git.bash

```diff
@@ -13,8 +13,13 @@ set -euo pipefail
 
 . misc/shlib/shlib.bash
 
-export BUILDKITE_REPO_REF="${BUILDKITE_REPO_REF:-origin}"
-export BUILDKITE_PULL_REQUEST_BASE_BRANCH="${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-main}"
+MZ_REPO_REF="${BUILDKITE_REPO_REF:-origin}"
+MZ_REPO_PULL_REQUEST_BASE_BRANCH="${BUILDKITE_PULL_REQUEST_BASE_BRANCH:-main}"
+
+if [[ "${BUILDKITE:-}" != "true" ]]; then
+  # when running locally, we origin may point to a fork of the repo but we want the mz repo
+  MZ_REPO_REF=$(git remote -v | grep "MaterializeInc/materialize" | grep "fetch" | head -n1 | cut -f1)
```

Comment:
Still not super happy with this line, for example someone could call themselves `FooMaterializeInc`, and then their fork would be used.

Reply:
Yes, as written the other solution proposed by you did not work, otherwise I would have gone with that.